src/shared/sleep-config.c

```diff
@@ -141,6 +144,134 @@ int read_battery_capacity_percentage(void) {
         return battery_capacity;
 }
 
+static int get_system_battery_identifier_hash(uint8_t ret[static SHA256_DIGEST_SIZE]) {
+        _cleanup_free_ char *bat_manufacturer = NULL, *bat_model_name = NULL, *bat_serial_number = NULL;
+        sd_id128_t machine_id, product_id;
+        struct sha256_ctx hash;
+        char *h = NULL, *p = NULL;
+        int r;
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/manufacturer", &bat_manufacturer);
+        if (r == -ENOENT)
+               log_debug_errno(r, "/sys/class/power_supply/BAT0/manufacturer is unavailable: %m");
+        else if (r < 0)
+               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/manufacturer: %m");
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/model_name", &bat_model_name);
+        if (r == -ENOENT)
+               log_debug_errno(r, "/sys/class/power_supply/BAT0/model_name is unavailable: %m");
+        else if (r < 0)
+               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/model_name: %m");
+
+        r = read_one_line_file("/sys/class/power_supply/BAT0/serial_number", &bat_serial_number);
+        if (r == -ENOENT)
+               log_debug_errno(r, "/sys/class/power_supply/BAT0/serial_number is unavailable: %m");
+        else if (r < 0)
+               return log_debug_errno(r, "Failed to read /sys/class/power_supply/BAT0/serial_number: %m");
+
+        r = sd_id128_get_machine(&machine_id);
+        if (r < 0)
+               return log_debug_errno(r, "Failed to get machine ID: %m");
+        h = SD_ID128_TO_STRING(machine_id);
```

Comment:
no need to change these to strings, given they are hashed, just pass a pointer to the sd_id128_t directly

Reply:
Need to check if it is not NULL

Reply:
no need, if the functions do not return an error, it is guaranteed to be set

Reply:
r = id128_get_product(&product_id);
        if (r == -ENOENT)
               log_debug_errno(r, "product_id does not exist: %m");

In case of -ENOENT, this would ignore and product_id would still be NULL. Converting NULL to sha256 would fail then

Reply:
ok Sorry i think i got what you meant. i can directly pass a pointer to sd_id128_t instead of converting it to string

Reply:
Thanks for the help. Fixed and pushed.